fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/DefaultOCFLObjectSession.java

```diff
@@ -393,6 +398,35 @@ public synchronized void close() throws PersistentStorageException {
         cleanupStaging();
     }
 
+    @Override
+    public List<VersionDetails> listVersions() throws PersistentStorageException {
+        assertSessionOpen();
+        //get a list of all versions in the object.
+        final List<VersionDetails> versionList = this.ocflRepository.describeObject(this.objectIdentifier)
+                                                                    .getVersionMap().values().stream()
+                                                                    // TODO uncomment when VersionDetails.isMutable()
+                                                                    // is available
+                                                                    //.filter(v -> !v.isMutable())
+                                                                    .collect(Collectors.toList());
+        //TODO remove following four lines once VersionDetails.isMutable() is available
+        //remove the last element in the list if it is a staged (ie not immutable) change.
+        if(this.ocflRepository.hasStagedChanges(this.objectIdentifier)){
+            versionList.remove(versionList.size()-1);
+        }
+
+        sort(versionList, VERSION_COMPARATOR);
```

Comment:
It appears that the staged/mutable head content would be considered a "mutable version", is this the only (current) example of a mutable version?
This is personal preference, but you could use your comparator to sort the list while streaming and then (once the `VersionDetails.isMutable()` stuff is finished return the list collected at line 410

Reply:
I'll look into that.